test-clients/src/main/java/com/hedera/services/bdd/spec/queries/HapiQueryOp.java

```diff
@@ -170,7 +172,9 @@ protected long feeFor(HapiApiSpec spec, Transaction txn, int numPayerKeys) throw
 	}
 
 	private Transaction fittedPayment(HapiApiSpec spec) throws Throwable {
-		if (nodePaymentFn.isPresent()) {
+		if (explicitPayment.isPresent()) {
+			return explicitPayment.get().signedTxnFor(spec);
```

Comment:
A silly question, queryHeader has a payment field of Transaction type. Is this explicitPayment use that field or it's a separate entity? How is the explicit payment transaction sent to services side if it's separate?

Reply:
@ljianghedera If we don't specify `explicitPayment` a default transaction is generated. Else it uses the payment transaction we provide.